AutoCollection/HttpDependencies.ts

```diff
@@ -60,48 +60,54 @@ class AutoCollectHttpDependencies {
         const originalHttpsRequest = https.request;
 
         const clientRequestPatch = (request: http.ClientRequest, options: string | URL | http.RequestOptions | https.RequestOptions) => {
-            var shouldCollect = !(<any>options)[AutoCollectHttpDependencies.disableCollectionRequestOption] &&
-                !(<any>request)[AutoCollectHttpDependencies.alreadyAutoCollectedFlag];
-
-            // If someone else patched traceparent headers onto this request
-            let userAgentHeader = null;
-
-            // Azure SDK special handling
-            if ((<any>options).headers) {
-                userAgentHeader = (<any>options).headers["User-Agent"] || (<any>options).headers["user-agent"];
-                if (userAgentHeader && userAgentHeader.toString().indexOf("azsdk-js") !== -1) {
-                    shouldCollect = false;
+            try {
+                var shouldCollect = !(<any>options)[AutoCollectHttpDependencies.disableCollectionRequestOption] &&
+                    !(<any>request)[AutoCollectHttpDependencies.alreadyAutoCollectedFlag];
+
+                // If someone else patched traceparent headers onto this request
+                let userAgentHeader = null;
+
+                // Azure SDK special handling
+                if ((<any>options).headers) {
+                    userAgentHeader = (<any>options).headers["User-Agent"] || (<any>options).headers["user-agent"];
+                    if (userAgentHeader && userAgentHeader.toString().indexOf("azsdk-js") !== -1) {
+                        shouldCollect = false;
+                    }
                 }
-            }
 
-            if (request && options && shouldCollect) {
-                CorrelationContextManager.wrapEmitter(request);
-                if (this._isEnabled) {
-                    // Mark as auto collected
-                    (<any>request)[AutoCollectHttpDependencies.alreadyAutoCollectedFlag] = true;
-
-                    // If there is no context create one, this apply when no request is triggering the dependency
-                    if (!CorrelationContextManager.getCurrentContext()) {
-                        // Create correlation context and wrap execution
-                        let operationId = null;
-                        if (CorrelationIdManager.w3cEnabled) {
-                            let traceparent = new Traceparent();
-                            operationId = traceparent.traceId;
+                if (request && options && shouldCollect) {
+                    CorrelationContextManager.wrapEmitter(request);
+                    if (this._isEnabled) {
+                        // Mark as auto collected
+                        (<any>request)[AutoCollectHttpDependencies.alreadyAutoCollectedFlag] = true;
```

Comment:
It was really just this line I was thinking about wrapping -- but that could cause repetitive requests. So this works too.